src/renovate.ts

```diff
@@ -1,39 +1,45 @@
 import Docker from './docker';
+import { Input } from './input';
 import { exec } from '@actions/exec';
 import fs from 'fs';
 import path from 'path';
 
 class Renovate {
-  private configFileEnv = 'RENOVATE_CONFIG_FILE';
-  private tokenEnv = 'RENOVATE_TOKEN';
   private dockerGroupName = 'docker';
   private configFileMountDir = '/github-action';
 
-  private configFile: string;
   private docker: Docker;
 
-  constructor(configFile: string, private token: string) {
-    this.configFile = path.resolve(configFile);
-
+  constructor(private input: Input) {
     this.validateArguments();
 
     this.docker = new Docker();
   }
 
   async runDockerContainer(): Promise<void> {
     const renovateDockerUser = 'ubuntu';
-    const githubActionsDockerGroupId = this.getDockerGroupId();
-    const commandArguments = [
-      '--rm',
-      `--env ${this.configFileEnv}=${this.configFileMountPath()}`,
-      `--env ${this.tokenEnv}=${this.token}`,
-      `--volume ${this.configFile}:${this.configFileMountPath()}`,
-      `--volume /var/run/docker.sock:/var/run/docker.sock`,
-      `--volume /tmp:/tmp`,
-      `--user ${renovateDockerUser}:${githubActionsDockerGroupId}`,
-      this.docker.image(),
-    ];
-    const command = `docker run ${commandArguments.join(' ')}`;
+
+    const dockerArguments = [
+      ...this.input.toEnvironmentVariables(),
+      {
+        key: this.input.options.configurationFile.env,
+        value: this.configFileMountPath(),
+      },
+    ]
+      .map((e) => {
+        const quotedValue = /\s/.test(e.value) ? `'${e.value}'` : e.value;
+        return `--env ${e.key}=${quotedValue}`;
```

Comment:
I think we should simply pass through existing env by `--env RENOVATE_XXX`. so no setting value.

Also the config file should be fully optional.

Reply:
> I think we should simply pass through existing env by `--env RENOVATE_XXX`. so no setting value.

I agree, except for the explicit token and configuration file ones, since they could be set using the input as well. Didn't know just doing `--env` was a thing to be honest.

> Also the config file should be fully optional.

Then the only way to get Renovate running would be by also adding the `RENOVATE_AUTODISCOVER=true` env right? If that is correct, wouldn't that do more harm than good as I'd guess most people would be using configured repositories in the config?

Reply:
Repositories can also be set via comma separated string to `RENOVATE_REPOSITORIES`

Reply:
Okay, I see. That isn't documented from what I could see at https://docs.renovatebot.com/self-hosting/, so thought autodiscover was the only way. Anyways I'll make the config file fully optional and pass all arguments except the token and configuration file using just `--env`.

Reply:
https://docs.renovatebot.com/self-hosted-configuration/#repositories

Generally all root renovate options can be set via env.

Reply:
Done!